File: AV.Common/Entities/Basket.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using AV.Contracts.Enums;

namespace AV.Common.Entities
{
    public class Basket: BaseEntity
    {
        public Guid CreatedBy { get; set; }
        public Guid UpdatedBy { get; set; }
        [Key]
        public int Id { get; set; }
        public Guid? PaymentId { get; set; }
        public string OrderNo { get; set; }
        public string PromoCode { get; set; }
        public Guid BuyerId { get; set; }
        public decimal NetTotal { get; set; }
        public decimal DiscountTotal { get; set; }
        public decimal GrossTotal { get; set; }
        public BasketStatus Status { get; set; }
        public List<BasketItem> Items { get; set; }
        public DateTimeOffset? ConfirmedOn { get; set; }

        public Basket()
        {
            Status = BasketStatus.Draft;
            CreatedDate = DateTimeOffset.UtcNow;
            UpdatedDate = DateTimeOffset.UtcNow;
            Items = new List<BasketItem>();
        }

        public Basket(Guid buyerId)
        {
            Items = new List<BasketItem>();
            BuyerId = buyerId;
            Status = BasketStatus.Draft;
            CreatedDate = DateTimeOffset.UtcNow;
            UpdatedDate = DateTimeOffset.UtcNow;
            CreatedBy = buyerId;
            UpdatedBy = buyerId;
        }

        public void RecalculateTotals()
        {
            NetTotal = Items.Sum(b => b.Price);
            DiscountTotal = Items.Sum(b => b.Discount);
            GrossTotal = NetTotal - DiscountTotal;
        }
    }
}

File: GoSmartValue.Web/ClientApp/src/app/shared/models/basket.ts
import {IBasket} from "./interfaces/basket";

export class Basket implements IBasket{
  "createdOn"= "0001-01-01T00:00:00+00:00";
  "updatedOn"= "0001-01-01T00:00:00+00:00";
  "id"= 0;
  "invoiceId"= 0;
  "orderNo"= "";
  "promoCode"= "";
  "buyerId"= "08d9c3ff-d554-4883-8609-a6d41a734dfd";
  "netTotal"= 0;
  "discountTotal"= 0;
  "grossTotal"= 0;
  "status"= 0;
  "items" = [
    {
      "createdDate": "2021-12-24T17:25:17.629867+00:00",
      "createdBy": "00000000-0000-0000-0000-000000000000",
      "updatedDate": "2021-12-24T17:25:17.629862+00:00",
      "updatedBy": "00000000-0000-0000-0000-000000000000",
      "id": 1,
      "basketId": 1,
      "productId": 1,
      "productName": "Standard 175",
      "unitPrice": 175,
      "discount": 0,
      "quantity": 1,
      "price": 1,
      "pictureUrl": "string",
      "promoCode": "string",
      "inputDataId": 0,
      "inputData": ""
    }
  ]
}
